shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.mm

```diff
@@ -4,20 +4,29 @@
 
 #include "flutter/fml/logging.h"
 
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterConstants.h"
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController_Internal.h"
 
-@implementation FlutterPlatformViewController
+@implementation FlutterPlatformViewController {
+  bool _embeddedViewsEnabled;
+}
 
 - (instancetype)init {
   self = [super init];
 
-  self->_platformViewFactories = [[NSMutableDictionary alloc] init];
+  _platformViewFactories = [[NSMutableDictionary alloc] init];
+  [self setEmbeddedViewsEnabled:[FlutterPlatformViewController embeddedViewsEnabled]];
   return self;
 }
 
 - (void)onCreateWithViewId:(int64_t)viewId
                   viewType:(nonnull NSString*)viewType
                     result:(nonnull FlutterResult)result {
+  if (!_embeddedViewsEnabled) {
```

Comment:
I don't think we need this, the `if (factory == nil)` already handles this. 
Same in `onDispose`